include/rapidjson/schema.h

```diff
@@ -1552,6 +1592,209 @@ struct TokenHelper<Stack, char> {
 
 } // namespace internal
 
+///////////////////////////////////////////////////////////////////////////////
```

Comment:
If this can be used out of schema, I think we may put it in separated file, similar to Pointer, regex. And then Schema depends on Uri and not vice versa. I have not read the schema yet. If URI is not part of public API, we may put it in internal folder.

Reply:
Yes I did consider that, but at the moment it uses the `GetXxxxString()` functions of `Schema` for convenience. It is exposed externally .... we have added a second flavour of resolver call, so that the resolver gets the Uri object, for convenience.

Reply:
Latest commit renames `Uri` to `GenericUri` and moves it to new file `uri.h`